docs/reference/modules/uploadfs.md

```diff
@@ -0,0 +1,359 @@
+        key: 'accessKey1',
+      },
+      {
+        account: 'storageAccountName2',
+        container: 'storageContainerName2',
+        key: 'accessKey2',
+      },
+    ]
+}
+```
+<template v-slot:caption>
+   modules/@apostrophecms/uploadfs/index.js
+</template>
+</AposCodeBlock>
+
+### `tempPath`
+During processing of files, the `uploadfs` module first copies them to a temporary location until the pipeline is finished. By default this is set to `data/temp/uploadfs` by the `node_modules/apostrophe/modules/@apostrophecms/uploadfs` module. The `tempPath` property takes a local directory path for creating the `/temp` folder that houses these files. It is deleted after the build process.
+
+## Featured methods
+
+### `copyIn(localPath, path, options, callback)`
+This method is used for handling uploads by the various storage types exposed by `uploadfs`. It is used by both the `asset` and `attachment` modules, but can also be used to handle files that aren't covered by the `attachment` module. The `copyIn()` method takes a local filename through the `localPath` argument. This file is copied to a uploadfs path specified by the `path` argument. Any necessary intermediate folders that don't exist will be created. The optional `options` argument is used to pass additional options to the selected storage. Finally, the callback receives any errors. It is often useful to wrap this function in  `promisify` to convert it to a promise-based function.
+
+#### Example
+```javascript
+async handleFile( file, uploadfsPath) {  
+  const copyIn = require('util').promisify(self.uploadfs.copyIn);
+  try {
+    await copyIn(file, uploadfsPath)
+  } catch (e) {
+    console.error(e);
+};  
+```
+
+### `copyOut(localPath, path, options, callback)`
+Much like the `copyIn` method, the `copyOut` method takes a `localPath` and a uploadfs `path` and copies the file back from uploadfs to the local file system. This is used when an existing asset needs to be updated, such as supplying additional cropping or other image processing. It should be used sparingly because it can result in poor storage service performance. Like the `copyIn` method, wrapping `copyOut` in `promisify` is often useful.
+
+#### Example
+
+```javascript
+async modifyImage( file, uploadfsPath) {  
+  const copyOut = require('util').promisify(self.uploadfs.copyOut);
+  const copyIn = require('util').promisify(self.uploadfs.copyIn);
+  try {  
+    await copyOut(file, uploadfsPath);
+    await newImageCrop(file);
+    await copyIn(file, uploadfsPath);
+   } catch (err) {
+    console.error(err);
+  }
+};
+```
+
+### `migrateFromDisabledFileKey(callback)`
+The `migrateFromDisabledFileKey` function provides a way to switch any file previously disabled using the `disabledFileKey` option to use the `disable` method instead. Note: the `disabledFileKey` option should be removed before invoking this method.
+
+#### Example
+
+```javascript
+self.uploadfs.migrateFromDisabledFileKey(function(e) { … });
```

Comment:
Come to that: we should document that `self.apos.uploadfs` is an actual uploadfs instance, the one used for attachments and also, by default, used for assets, and that you can use it too for custom uploads as long as you avoid the prefixes `/attachments` and `/apos-frontend` and pick a prefix directory name of your own for your custom uploadfs paths.

The `@apostrophecms/uploadfs` module itself has no convenience alias to access it via `self.apos.*` because the shared uploadfs asset is more useful and what the developer probably wants. We made the same call for `self.apos.express` (this is the `express` instance, not the `@apostrophecms/express` module) and`self.apos.db` (this is the MongoDB db object, not the `@apostrophecms/db` module). In rare cases where you really want to access one of these "factory" modules and not the service it provides, you can write `self.apos.modules['@apostrophecms/uploadfs']`, and in particular you'll use this syntax if you do decide you need to call `getInstance` to create an entirely separate `uploadfs` instance for something.

Reply:
Altered the examples.
Not sure I fully get your second comment. I added a paragraph in the introductory section that is a distillation of what you said.
So I think I get the difference between `self.apos.uploadfs` and `self.apos.modules['@apostrophecms/uploadfs']`, but I don't quite get the use case. I'll have to think about it/maybe make up a fake case for myself.